modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+		label: '1 Page title',
+		name: '0',
+		value: '0',
+	},
+	{
+		label: '2 Page title',
+		name: '1',
+		value: '1',
+	},
+	{
+		label: '3 Page title',
+		name: '2',
+		value: '2',
+	},
+];
+
+const conditions = [
+	{
+		operands: [
+			{
+				type: '',
+				value: '',
+			},
+		],
+		operator: '',
+	},
+	{
+		operands: [
+			{
+				type: '',
+				value: '',
+			},
+		],
+		operator: '',
+	},
+];
+
+const newRule = {
+	...DEFAULT_RULE,
+	conditions,
+};
+
+const defaultProps = (fieldsList = FIELDS) => {
+	return {
+		allowActions: [
+			'auto-fill',
+			'calculate',
+			'enable',
+			'jump-to-page',
+			'require',
+			'show',
+		],
+		dataProvider: [
+			{
+				id: '39421',
+				label: 'Get countries',
+				name: 'Get countries',
+				uuid: 'a6ab90fd-c4ac-b91b-c9a5-cf756170d9d3',
+				value: '39421',
+			},
+		],
+		fields: fieldsList,
+		operatorsByType: OPERATORS_BY_TYPE,
+		pages,
+		rule: DEFAULT_RULE,
+	};
+};
+
+describe('Editor', () => {
+	const originalLiferayLoader = window.Liferay.Loader;
+	afterEach(() => {
+		cleanup();
+	});
+
+	beforeEach(() => {
+		global.fetch.mockResponse(
+			JSON.stringify([
+				{
+					javaScriptModule: 'checkbox_multiple',
+					name: 'checkbox_multiple',
+				},
+				{
+					javaScriptModule: 'color',
+					name: 'color',
+				},
+				{
+					javaScriptModule: 'date',
+					name: 'date',
+				},
+				{
+					javaScriptModule: 'image',
+					name: 'image',
+				},
+				{
+					javaScriptModule: 'numeric',
+					name: 'numeric',
+				},
+				{
+					javaScriptModule: 'select',
+					name: 'select',
+				},
+				{
+					javaScriptModule: 'text',
+					name: 'text',
+				},
+				{
+					javaScriptModule: 'richText',
+					name: 'richText',
+				},
+				{
+					javaScriptModule: 'grid',
+					name: 'grid',
+				},
+				{
+					javaScriptModule: 'document_library',
+					name: 'document_library',
+				},
+			])
+		);
+	});
+
+	beforeAll(() => {
+		window.Liferay = {
+			...window.Liferay,
+			Loader: {
+				require: ([fieldModule], resolve) => {
+					switch (fieldModule) {
+						case 'color':
+							resolve({default: Color});
+							break;
+						case 'date':
+							resolve({default: Date});
+							break;
+						case 'grid':
+							resolve({default: Grid});
+							break;
+						case 'image':
+							resolve({default: Image});
+							break;
+						case 'numeric':
+							resolve({default: Numeric});
+							break;
+						case 'richText':
+							resolve({default: RichText});
+							break;
+
+						case 'select':
+							resolve({default: Select});
+							break;
+						case 'text':
+							resolve({default: Text});
+							break;
+						case 'document_library':
+							resolve({default: DocumentLibrary});
+							break;
+						default:
+							break;
+					}
+				},
+			},
+		};
+	});
+
+	afterAll(() => {
+		window.Liferay.Loader = originalLiferayLoader;
+	});
+
+	describe('Editor', () => {
+		describe('Conditions', () => {
+			describe('Field operator', () => {
+				it.each(STRING_DATATYPE_FIELDS)(
```

Comment:
`				it.each(STRING_DATATYPE_FIELDS)(`
![image](https://user-images.githubusercontent.com/7663513/104361984-fccbd380-54f1-11eb-81f2-319cd27a1187.png)